### PR TITLE
Replaces `String socketPath`  by an Endpoint

### PR DESCRIPTION
**Pull request by CarlosNihelton**
Hello! This is the first attempt to flexibilize the relationship between the Flutter GUI clients and Subiquity server. Replacing the socket paths passed around in the API surface by a higher level construct enables us soon to compose other connections, like TCP, without changing the API.

### CODE REVIEW

File: packages/subiquity_client/lib/src/server.dart [hunk 1]
```diff
@@ -11,6 +11,25 @@ enum ServerMode { LIVE, DRY_RUN }
 /// @internal
 final log = Logger('subiquity_server');
 
+/// Vocabulary type that combines an [InternetAddress] and a [port] number
+class Endpoint {
```
Comment (jpnurmi) [inline packages/subiquity_client/lib/src/server.dart:15]: could we have the `Endpoint` class in a separate `endpoint.dart` file so the client and server don't need to import each other?

File: packages/subiquity_client/lib/src/server.dart [hunk 2]
```diff
@@ -11,6 +11,25 @@ enum ServerMode { LIVE, DRY_RUN }
...
+
+  /// Creates an Unix domain socket endpoint.
+  factory Endpoint.unix(String socketPath) => Endpoint(
+        InternetAddress(socketPath, type: InternetAddressType.unix),
+        port: 0,
+      );
+
+  /// Creates a localhost TCP endpoint on port number [port].
+  factory Endpoint.localhost(int port) => Endpoint(
+        InternetAddress("127.0.0.1", type: InternetAddressType.IPv4),
```
Comment (jpnurmi) [inline packages/subiquity_client/lib/src/server.dart:27]: i thought we had prefer_single_quotes enabled :D

File: packages/subiquity_client/lib/src/client.dart [hunk 3]
```diff
@@ -41,23 +41,22 @@ class SubiquityException implements Exception {
...
+  late Endpoint _endpoint;
   final _isOpen = Completer<bool>();
 
   Future<bool> get isOpen => _isOpen.future;
 
-  void open(String socketPath) {
-    log.info('Opening socket ${p.absolute(socketPath)}');
-    _socketPath = socketPath;
+  void open(Endpoint endpoint) {
+    log.info('Opening socket address ${endpoint.address}');
```
Comment (jpnurmi) [inline packages/subiquity_client/lib/src/client.dart:50]: we could even implement Endpoint.toString()? :)

**Review by jpnurmi — COMMENTED** [review]
Some nitpick but overall looks good! :+1:

File: packages/subiquity_client/lib/src/server.dart (hunk 1 above)
Comment (CarlosNihelton) [reply packages/subiquity_client/lib/src/server.dart]: Silly me not noticing the server was imported into the client automatically. Sorry about that. Fixed now.

File: packages/subiquity_client/lib/src/server.dart (hunk 2 above)
Comment (CarlosNihelton) [reply packages/subiquity_client/lib/src/server.dart]: That deserves a PR on its own :D

image

**Review by jpnurmi — APPROVED** [review]
LGTM :+1:

**Pull request merged**